### ISSUE TITLE
Pre-commit hook fails on windows

### ISSUE THREAD

**Issue by VakarisZ** [pre-commit/mirrors-eslint#31]
As soon as I add this hook to `.pre-commit-config.yaml`:
```
default_stages: [commit]
repos:
  - repo: https://github.com/pycqa/isort
    rev: 5.8.0
    hooks:
      - id: isort
        name: isort (python)
      - id: isort
        name: isort (cython)
        types: [cython]
      - id: isort
        name: isort (pyi)
        types: [pyi]
  - repo: https://github.com/eslint/eslint
    rev: v7.24.0
    hooks:
      - id: eslint
  - repo: https://github.com/psf/black
    rev: 20.8b1
    hooks:
      - id: black
  - repo: https://gitlab.com/pycqa/flake8
    rev: 3.9.1
    hooks:
      - id: flake8
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v3.4.0
    hooks:
      - id: check-added-large-files
      - id: check-case-conflict
      - id: check-json
      - id: check-merge-conflict
      - id: detect-private-key
      - id: end-of-file-fixer
      - id: trailing-whitespace
  - repo: local
    hooks:
      - id: pytest
        name: pytest
        entry: bash -c "cd monkey && pytest"
        language: system
        files: "monkey/"
        exclude: "monkey/monkey_island/cc/ui"
        stages: [push]
```

When I run pre-commit run:
```
[INFO] Installing environment for https://github.com/eslint/eslint.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: FileNotFoundError: [WinError 3] The system cannot find the path specified: 'C:\\Users\\Vakaris\\.cache\\pre-commit\\repotvkewz60\\node_modules\\eslint\
\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_mod
ules\\eslint\\node_modules'
Check the log at C:\Users\Vakaris\.cache\pre-commit\pre-commit.log
```

The log:

### version information

```
pre-commit version: 2.12.1
sys.version:
    3.7.7 (tags/v3.7.7:d7c567b08f, Mar 10 2020, 10:41:24) [MSC v.1900 64 bit (AMD64)]
sys.executable: c:\users\vakaris\desktop\test_venv\scripts\python.exe
os.name: nt
sys.platform: win32
```

### error information

```
An unexpected error has occurred: FileNotFoundError: [WinError 3] The system cannot find the path specified: 'C:\\Users\\Vakaris\\.cache\\pre-commit\\repotvkewz60\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules'
```

```
Traceback (most recent call last):
  File "c:\users\vakaris\desktop\test_venv\lib\site-packages\pre_commit\error_handler.py", line 65, in error_handler
    yield
  File "c:\users\vakaris\desktop\test_venv\lib\site-packages\pre_commit\main.py", line 385, in main
    return run(args.config, store, args)
  File "c:\users\vakaris\desktop\test_venv\lib\site-packages\pre_commit\commands\run.py", line 406, in run
    install_hook_envs(hooks, store)
  File "c:\users\vakaris\desktop\test_venv\lib\site-packages\pre_commit\repository.py", line 224, in install_hook_envs
    _hook_install(hook)
  File "c:\users\vakaris\desktop\test_venv\lib\site-packages\pre_commit\repository.py", line 83, in _hook_install
    hook.prefix, hook.language_version, hook.additional_dependencies,
  File "c:\users\vakaris\desktop\test_venv\lib\site-packages\pre_commit\languages\node.py", line 117, in install_environment
    rmtree(prefix.path('node_modules'))
  File "c:\users\vakaris\desktop\test_venv\lib\site-packages\pre_commit\util.py", line 265, in rmtree
    shutil.rmtree(path, ignore_errors=False, onerror=handle_remove_readonly)
  File "C:\Users\Vakaris\AppData\Local\Programs\Python\Python37\lib\shutil.py", line 516, in rmtree
    return _rmtree_unsafe(path, onerror)
  File "C:\Users\Vakaris\AppData\Local\Programs\Python\Python37\lib\shutil.py", line 395, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "C:\Users\Vakaris\AppData\Local\Programs\Python\Python37\lib\shutil.py", line 395, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "C:\Users\Vakaris\AppData\Local\Programs\Python\Python37\lib\shutil.py", line 395, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  [Previous line repeated 17 more times]
  File "C:\Users\Vakaris\AppData\Local\Programs\Python\Python37\lib\shutil.py", line 377, in _rmtree_unsafe
    onerror(os.scandir, path, sys.exc_info())
  File "C:\Users\Vakaris\AppData\Local\Programs\Python\Python37\lib\shutil.py", line 374, in _rmtree_unsafe
    with os.scandir(path) as scandir_it:
FileNotFoundError: [WinError 3] The system cannot find the path specified: 'C:\\Users\\Vakaris\\.cache\\pre-commit\\repotvkewz60\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules\\eslint\\node_modules'
```

**Comment by asottile** [comment]
can you try with a newer version of python?  I seem to remember this being a bug in python3.7 that they never fixed in that version (it's recursing on a symlink)

**Comment by VakarisZ** [comment]
You're right, it works on 3.9
